qcore/utils.py

```diff
@@ -123,20 +123,36 @@ def load_sim_params(sim_yaml_path, load_fault=True, load_root=True, load_vm=True
     :param load_vm: to load vm_params.yaml or not
     :return: a DotDictify object that contains all necessary params for a single simulation
     """
-    sim_params = load_yaml(sim_yaml_path)
+    sim_params = {}
     fault_params = {}
     root_params = {}
     vm_params = {}
-    if load_root or load_vm and not load_fault:
+
+    if load_root is True or load_vm is True and not load_fault:
         load_fault = True  # root/vm_yamlpath in fault_yaml
-    if load_fault:
+
+    if sim_yaml_path:
+        sim_params = load_yaml(sim_yaml_path)
+    elif load_fault is True:
+        raise ValueError("For automated fault_params loading, sim_params must be set")
```

Comment:
Isn't sim_yaml_path always set as there is no default?

Reply:
Good question.  sim_params.yaml doesn't exist before INSTALL_REL runs, so before then we have to load the individual params yamls somewhat manually.
Happy to take input on an improved method